web/src/lib/components/shared-components/search-bar/search-filter-box.svelte

```diff
@@ -208,18 +219,99 @@
         make: undefined,
         model: undefined,
       },
-      dateRange: {
-        startDate: undefined,
-        endDate: undefined,
+      date: {
+        takenAfter: undefined,
+        takenBefore: undefined,
       },
-      inArchive: undefined,
-      inFavorite: undefined,
-      notInAlbum: undefined,
+      isArchive: undefined,
+      isFavorite: undefined,
+      isNotInAlbum: undefined,
       mediaType: MediaType.All,
     };
   };
 
-  const search = () => {};
+  const search = async () => {
+    let type: AssetTypeEnum | undefined = undefined;
+
+    if (filter.mediaType === MediaType.Image) {
+      type = AssetTypeEnum.Image;
+    } else if (filter.mediaType === MediaType.Video) {
+      type = AssetTypeEnum.Video;
+    }
+
+    let payload: SmartSearchDto | MetadataSearchDto = {
+      country: filter.location.country?.value,
+      state: filter.location.state?.value,
+      city: filter.location.city?.value,
+      make: filter.camera.make?.value,
+      model: filter.camera.model?.value,
+      takenAfter: filter.date.takenAfter
+        ? DateTime.fromFormat(filter.date.takenAfter, 'yyyy-MM-dd').toUTC().startOf('day').toString()
+        : undefined,
+      takenBefore: filter.date.takenBefore
+        ? DateTime.fromFormat(filter.date.takenBefore, 'yyyy-MM-dd').toUTC().endOf('day').toString()
+        : undefined,
+      isArchived: filter.isArchive,
+      isFavorite: filter.isFavorite,
+      isNotInAlbum: filter.isNotInAlbum,
+      personIds: filter.people && filter.people.length > 0 ? filter.people.map((p) => p.id) : undefined,
+      type,
+    };
+
+    if (filter.context) {
+      if (payload.personIds && payload.personIds.length > 0) {
+        handleError(
+          new Error('Context search does not support people filter'),
+          'Context search does not support people filter',
+        );
+        return;
+      }
+
+      payload = {
+        ...payload,
+        query: filter.context,
+      };
+    }
+
+    dispatch('search', payload);
+  };
+
+  function populateExistingFilters() {
+    console.log($searchQuery?.takenAfter);
+    if ($searchQuery) {
+      filter = {
+        context: 'query' in $searchQuery ? $searchQuery.query : '',
+        people:
+          'personIds' in $searchQuery ? ($searchQuery.personIds?.map((id) => ({ id })) as PersonResponseDto[]) : [],
+        location: {
+          country: $searchQuery.country ? { label: $searchQuery.country, value: $searchQuery.country } : undefined,
```

Comment:
Since `$searchQuery.xyz` can either be undefined or set, something like `$searchQuery.country && { label: $searchQuery.country, value: $searchQuery.country }` should yield the same result

Reply:
Unfortunately it doesn't because it `$searchQuery.country` doesn't return the `{label: string, value: string}` type

Reply:
It doesn't have to, right? The idea is to set the object if $seachQuery.country is defined, otherwise it will be undefined.